cmd/fleet/serve_test.go

```diff
@@ -161,3 +166,112 @@ func TestCronWebhooks(t *testing.T) {
 	time.Sleep(1 * time.Second)
 	assert.GreaterOrEqual(t, int32(2), atomic.LoadInt32(&endpointCalled))
 }
+
+func TestCronVulnerabilitiesCreatesDatabasesPath(t *testing.T) {
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+	ds := new(mock.Store)
+	ds.AppConfigFunc = func() (*fleet.AppConfig, error) {
+		return &fleet.AppConfig{}, nil
+	}
+
+	vulnPath := path.Join(t.TempDir(), "something")
+	require.NoDirExists(t, vulnPath)
+
+	fleetConfig := config.FleetConfig{
+		Vulnerabilities: config.VulnerabilitiesConfig{
+			DatabasesPath:         vulnPath,
+			Periodicity:           10 * time.Second,
+			CurrentInstanceChecks: "auto",
+		},
+	}
+	go cronVulnerabilities(ctx, ds, kitlog.NewNopLogger(), &alwaysLocker{}, "AAA", fleetConfig)
+
+	time.Sleep(1 * time.Second)
```

Comment:
It would be nice to avoid `time.Sleep` in tests when possible - it often makes them flaky/racy or unnecessarily slow. I think in this case we could take advantage of the context, and using something like this (instead of the go and sleep calls), you wouldn't wait any more than necessary:

```
// I make it 100ms, but it could even already be expired in this case since the
// tested behaviour is before the timer-based loop.
ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
defer cancel()

var wg sync.WaitGroup
wg.Add(1)

go func() {
  defer wg.Done()
  cronVulnerabilities(...) // same args
}()
wg.Wait()
```

Reply:
Agreed! Thank you.

Reply:
Addressing this made me realize the exit portion of the crons was wrong, we were breaking the select, not the for.

Reply:
Indeed! Good catch.